ports/unix/mpthreadport.c

```diff
@@ -56,6 +59,18 @@ STATIC thread_t *thread;
 // it's needed because we can't use any pthread calls in a signal handler
 STATIC sem_t thread_signal_done;
 
+#if defined (__APPLE__)
+STATIC char thread_semaphore_name[3] = {"tsn"};
+#define GEN_SEM_NAME(sem_name)                              \
```

Comment:
This random name is probably not going to work well enough.  There are 2 things to protect against: 1) another program using the same name; 2) another MicroPython process running at the same time.  A way to do it would be to have the name explicitly MicroPython, and with the process number attached at the end, eg `"micropython_sem_12345"` (where 12345 is the process number which is dynamically updated).